QuantConnect.TradierBrokerage/TradierBrokerage.cs

```diff
@@ -1324,83 +1287,17 @@ private void ProcessPotentiallyUpdatedOrder(TradierCachedOpenOrder cachedOrder,
-                                            // if we failed to place this order I don't know what to do, we've filled the first part
-                                            // and failed to place the second... strange. Should we invalidate the rest of the order??
-                                            Log.Error("TradierBrokerage.SubmitContingentOrder(): Failed to submit contingent order.");
-                                            var message = $"{qcOrder.Symbol} Failed submitting contingent order for " +
-                                                $"QC id: {qcOrder.Id.ToStringInvariant()} Filled " +
-                                                $"Tradier Order id: {updatedOrder.Id.ToStringInvariant()}";
-                                            OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "ContingentOrderFailed", message));
-                                            OnOrderEvent(new OrderEvent(qcOrder, DateTime.UtcNow, orderFee) { Status = OrderStatus.Canceled });
-                                        }
-                                    }
-                                    catch (Exception err)
-                                    {
-                                        Log.Error(err);
-                                        OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "ContingentOrderError", "An error occurred while trying to submit an Tradier contingent order: " + err));
-                                        OnOrderEvent(new OrderEvent(qcOrder, DateTime.UtcNow, orderFee) { Status = OrderStatus.Canceled });
-                                    }
-                                    finally
-                                    {
-                                        _contingentReentranceGuardByQCOrderID.Remove(qcOrder.Id);
-                                    }
-                                });
-                            }
-                        }
-                    }
-
-                    // zero-crossing order completely filled, remove entry
-                    var closingOrderId = _zeroCrossingOrdersByTradierClosingOrderId.FirstOrDefault(x => qcOrder.Id == x.Value.Id).Key;
-                    if (closingOrderId > 0 && closingOrderId != updatedOrder.Id)
-                    {
-                        Order removed;
-                        _zeroCrossingOrdersByTradierClosingOrderId.Remove(closingOrderId, out removed);
-                    }
+                    OnOrderEvent(fill);
                 }
-
-                OnOrderEvent(fill);
             }
 
             // remove from open orders since it's now closed
             if (OrderIsClosed(updatedOrder))
             {
                 _filledTradierOrderIDs.Add(updatedOrder.Id);
-                _cachedOpenOrdersByTradierOrderID.TryRemove(updatedOrder.Id, out cachedOrder);
+                _cachedOpenOrdersByTradierOrderID.TryRemove(updatedOrder.Id, out _);
```

Comment:
`Remove` instead of `TryRemove`, review it I think there's an overload that doesn't return `out`

Reply:
The `ConcurrentDictionary` doesn't have `Remove()` explicitly.
![image](https://github.com/QuantConnect/Lean.Brokerages.Tradier/assets/45608740/2bb07fea-afe2-48f8-8c60-9ecdd8ed0d4d)

We can cast to `IDictionary<TKey, TValue>` collection to call `Remove()`. Does it make sense ?
![image](https://github.com/QuantConnect/Lean.Brokerages.Tradier/assets/45608740/bd8c74cf-52e7-4e0a-a8a2-b98dd22dc192)